library-valcon-internship/src/components/Modals/ModalAddAuthor/ModalAddAuthor.tsx

```diff
@@ -0,0 +1,124 @@
+import { Dispatch, FormEvent, SetStateAction, SyntheticEvent, useState } from 'react'
+
+import ReactDOM from 'react-dom'
+
+import { addNewAuthor } from '../../../services/AuthorService'
+import '../Modals.css'
+import './ModalAddAuthor.css'
+
+interface AddAuthorProps {
+  show: boolean,
+  closeModal: () => void,
+  retrieveAuthors: boolean,
+  setRetrieveAuthors: Dispatch<SetStateAction<boolean>>
+}
+
+const ModalAddAuthor = ({ show, closeModal, retrieveAuthors, setRetrieveAuthors }: AddAuthorProps) => {
+  const [ firstName, setFirstName ] = useState('')
+  const [ lastName, setLastName ] = useState('')
+  const [ invalidFirstName, setInvalidFirstName ] = useState(false)
+  const [ invalidLastName, setInvalidLastName ] = useState(false)
+  const [ invalidAuthorData, setInvalidAuthorData ] = useState(false)
```

Comment:
Why are we assuming the input is valid on init? Maybe it would be better not to use negation in boolean names.
eg: isFirstNameValid, isLastNameValid, isAuthorDataValid.

Reply:
setInvalidFirstName sounds like we are setting firstName to some invalid value